packages/ra-ui-materialui/src/input/AutocompleteArrayInput.js

```diff
@@ -104,25 +104,27 @@ const styles = theme => ({
 export class AutocompleteArrayInput extends React.Component {
     state = {
         dirty: false,
-        inputValue: null,
+        inputValue: [],
         searchText: '',
         suggestions: [],
     };
 
     inputEl = null;
 
+    getInputValue = inputValue => (inputValue === '' ? [] : inputValue);
```

Comment:
I think you should refactor the empty array to a constant here, otherwise the function will return a *new* empty array at each call, and this will force a rerender of teh AutocompleteArrayInputChip each time the AutocompleteArrayInput is rerendered.

And while you are at it, use the same constant in the initial state.